### DISCUSSION TITLE
User und Devices zum Testen

### DISCUSSION THREAD

**Discussion by emufan** [SukramJ/aiohomematic#127] [Ideas]
GIbt ja immer mal wieder was zum Testen/Reproduzieren, usw., wie aktuell im BROLL-Fall und nicht jedes hat ja alles. Daher die Idee hier mal zu sammeln, sodass man bei Problemen weiß, wer mittesten/reproduzieren könnte.

Mein Fuhrpark ist eher klein, da ich erst 2021 angefangen habe, aber zumindest kann ich folgende beisteuern

```
CCU3 Original
+ HmIP-HAP

HmIP-BROLL
HmIP-BSM
HmIP-BWTH
HmIP-DRSI1
HmIP-eTRV-C-2
HmIP-FCI1
HmIP-FSM16
HMIP-PSM
HmIP-PSM-2
HmIP-SLO
HmIP-SMI
HmIP-SRH
HmIP-STHD
HmIP-STHO
HmIP-SWD
HMIP-SWDO
HmIP-SWDO-I
HMIP-WRC2
HmIP-WRC6
HmIP-WTH-2

HM-ES-TX-WM

HmIP-HEATING
```

**Comment by itn3rd77** [comment]
Hi,

gute Idee. Wollte ich auch schon anfangen. Sollten wir evtl im Wiki machen, dass ist dann einfacher zu pflegen und wir könnten auch den Status pro Gerät aufnehmen z.B. (vollständig getestet etc.).

Ich kann noch folgende beisteuern:

HmIP-eTRV-2
HmIP-FSM
HmIP-RC8
HmIP-STHO-A
HMIP-WRC2

HM-CC-RT-DN
HM-LC-Bl1PBU-FM
HM-LC-Dim1TPBU-FM

**Reply by emufan** [reply]
Dachte, Du hättest auch ein BROLL? Meine Idee war, dass Daniel, SukramJ oder User mit Problemen in der aktuellen Phase gucken können, wer bereit wäre zum Testen und nicht nur alle disjunkten Systemen zu sammeln. So wie gestern mit dem aktuellen BROLL Problem, wo er ja nur zufällig wusste, dass Du das hast und Dich um Mittesten gebeten hat, es von mir aber nicht wusste, usw.

**Reply by danielperna84** [reply]
Ich stimme zu, dass hier jeder am besten eine eigene Antwort postet mit seinen Geräten. Da kann man dann auch selbst schnell mal seine Antwort anpassen wenn neue Geräte dazu gekommen sind. Das zentral zu pflegen ist bei (evtl.) zahlreichen Teilnehmern vermutlich nicht mehr so einfach zu pflegen. 🤔

**Comment by danielperna84** [comment]
Super Sache. 👍

Wenn ich an diese Stelle noch etwas ergänzen darf: ich sammle in meinem Projekt [pydevccu](https://github.com/danielperna84/pydevccu) die ganzen Datenpunkte die mir die User schon für pyhomematic haben zukommen lassen. Damit ist es bei der Entwicklung möglich zumindest rein auf Parameter-Ebene Geräte zu simulieren. Bzw. lässt sich auch die Funktion _simulieren_, aber das ist nur für 1-2 Geräte beispielhaft implementiert.

Wie dem auch sei, wenn ihr Geräte habt die dort fehlen, dann könnt ihr dort gerne Pull Requests öffnen um die _Datenbank_ zu vervollständigen. Irgendwann mal wird das dann ggf. genutzt um automatisierte Tests von hahomematic zu machen. Und bis dahin kann man als Entwickler zumindest so weit implementieren, dass die Devices / entities den Erwartungen entsprechend von hahomematic erzeugt werden.

Vielleicht noch direkt dazu schreiben, welche Parameter / entities in euren Augen relevant für das jeweilige Gerät sind. Gerade die von HmIP bringen viele Datenpunkte mit, die oft in der Praxis irrelevant sind. Dann ggf. noch ein Screenshot dazu wie das in dem CCU UI aussieht, dann hat man eine gute Basis um ein Gerät zumindest _theoretisch_ zu implementieren. 👍

Wer das Skript nicht schon kennt: [get_device_data.py](https://github.com/danielperna84/pydevccu/blob/master/pydevccu/get_device_data.py) erzeugt pro Gerät immer 2 JSON-Dateien. Die Device-Description enthält erst mal nur grundsätzlich die Liste der Kanäle, die Paramset-Description dann die eigentlichen Parameter auf der Kanälen. Damit das mit pydevccu funktioniert müssen beide Dateien vorhanden sein. Am besten einfach mal in die bestehenden Dateien rein schauen um eine Idee davon zu bekommen.

Meine Geräte (via Debmatic auf einem Pi3 mit RPI-RF-MOD):
```
HM-CC-RT-DN
HM-ES-TX-WM (mit dem LED-Sensor)
HM-LC-Sw1-FM
HM-LC-Bl1PBU-FM
HM-RC-Key4-3
HM-Sec-Key
HM-Sec-SC-2
HM-Sen-MDIR-WM55
HM-TC-IT-WM-W-EU
ZEL STG RM FEP 230V
HmIP-BDT
HmIP-DRSI1
HmIP-DRSI4
HmIP-eTRV-B
HmIP-PSM
HmIP-SMI55
HmIP-SWDO-I
HmIP-SWSD
```

**Reply by SukramJ** [reply]
Ich hab in Homematic(IP) Local auch einen Service eingebaut, der die benötigten Dateien in den Order` {ha-config}/hahm/...` exportiert.
Ist hoffentlich eine Vereinfachung im vergleich zur Scriptdatei.

**Reply by SukramJ** [reply]
Der Service heißt: `Homematic(IP) Local: Export device definition`

**Comment by Lemocin** [comment]
Hi, 
klasse Idee, mir kam gestern Abend die gleiche :D 
Aktuell habe ich auch eher weniger Geräte. Die wären: 
```
HmIP-eTRV-B
HmIP-eTRB-B1
HmIP-BROLL
HmIP-SWDM-B2
HmIP-SWDM
CCU3 mit einer piVCCU(oDroid C2, HM-MOD-RPI-PCB)
```
Ich denke es könnte auch interessant sein, welche CCU auf welchem Gerät man verwendet. Das habe ich hier mal ergänzt.

**Comment by noxhirsch** [comment]
Top Idee - hier meine Liste:

```
RaspberryMatic auf Synology NAS + RPI-RF-MOD
HmIP-HAP

HMIP-SWDO
HmIP-SWDO-I
HMIP-WRC2
HmIP-WRC6
HmIP-FCI1
HmIP-FCI6
HmIP-MOD-RC8
HmIP-MOD-OC8
HmIP-SPI
HmIP-SMI
HmIP-eTRV-2
HmIP-eTRV-E
HmIP-STH
HmIP-SWO-PL
HmIP-DSD-PCB
HMIP-PS
HMIP-PSM
HmIP-SWD
HmIP-SLO
HmIP-STHO
HmIP-USBSM
HmIP-WRCR
HmIP-SCI

HM-CC-RT-DN
HM-WDS30-OT2-SM
HM-Sec-SCo
HM-LC-Sw1-FM
HM-LC-Dim1L-CV
HM-ES-PMSw1-Pl-DN-R1

(HmIP-HEATING)
(HM-CC-VG-1)

In Kürze:
HmIP-BSL
```

**Comment by eikowagenknecht** [comment]
Meine Devices, auf CCU3 mit Raspberrymatic und externer Quad Antenne von stall.biz:

HM-LC-Bl1-FM
HM-LC-Sw1-FM
HM-LC-Sw4-DR
HM-PBI-4-FM
HM-Sen-RD-O

HmIP-DLD
HmIP-DRSI4
HmIP-DSD-PCB
HmIP-eTRV-2
HmIP-FCI1
HmIP-FCI6
HMIP-PSM
HmIP-SCTH230
HmIP-SPI
HMIP-SWDO
HmIP-SWDO-I
HmIP-SWO-PR
HmIP-SWSD
HmIP-WRCD

**Reply by SukramJ** [reply]
@eikowagenknecht funktioniert eigentlich der HmIP-DLD? lock, unlock, open

**Reply by eikowagenknecht** [reply]
@SukramJ Ich habe es tatsächlich noch nicht verbaut, aber auf meinem Schreibtisch stehend scheint es über die Geräte Ansteuerung alle 3 Befehle /auf/zu/öffnen korrekt auszuführen. Dabei meldet es in Homematic einen "ERROR_JAMMED=TRUE", was auch Sinn macht, da es nicht schließen kann. Wie bekomme ich dieses Event in Home Assistant mit?

**Reply by SukramJ** [reply]
Kann ich dir nicht sagen, `ERROR_JAMMED` kann ich in der Doku nicht finden.
Vielleicht  mal mit dem Homematic Manager drauf schauen.

**Reply by eikowagenknecht** [reply]
Dokumentiert oder nicht, es wird gesendet auf Kanal 0. Hier der Ausschnitt aus dem Homematic Manager: 
![image](https://user-images.githubusercontent.com/1475672/149345918-879f34aa-5fc1-401f-8b68-d7590a049dc6.png)
Es wäre großartig, wenn das irgendwie in Home Assistant reinkönnte, z.B. als eigene Entität, oder auch als Event, egal. Hauptsache ich kann drauf reagieren und einen Fehler ausgeben wenn das Abschließen nicht geklappt hat. Das ist m.E. eine wichtige Funktionalität für ein Türschloss, das ja Sicherheit bieten soll..

**Reply by SukramJ** [reply]
Ohne Doku kann ich das Ding halt keinem Kanal zuordnen
Ich würde daraus eine persistent_notification machen.

**Reply by eikowagenknecht** [reply]
Könnte es nicht besser als Entität dargestellt werden, genau wie LOW_BAT z.B. ja auch? Dann kann man direkter darauf reagieren und muss nicht die Notifications parsen. Von den automatisch erstellten Notifications bin ich generell eher kein Freund (siehe die andere Diskussion zu dem Thema hier).

Benötigst du noch irgendwelche Debugdaten von mir dazu?

**Reply by SukramJ** [reply]
Also das mit der Doku ist ein Problem für mich, und auch die in pydevccu eingecheckten paramsets und device descriptions enthalten den Parameter nicht. Kann Du bitte mit dem Service` hahm.export_device_definition` einen Export machen? Vielleicht gibt es da mittlerweile eine V2 des Gerätes. Die Dateien liegen dann in deinem HA-Config Verzeichnis unter hahm. Im paramset sollte der Parameter dann auftauchen.

**Reply by SukramJ** [reply]
Die Entität würde ich sowieso erstellen und an den persisten_notifications mach ich vorerst nichts, war ein kleiner Scherz.

**Reply by eikowagenknecht** [reply]
Ah, da war mein Ironiedetektor wohl kaputt oder dein Humor ist trockener als das was ich gewohnt bin ;-)

Hier die HmIP-DLD.json:
[hahm.zip](https://github.com/danielperna84/hahomematic/files/7863549/hahm.zip)

```
      "ERROR_JAMMED": {
        "MIN": false,
        "OPERATIONS": 5,
        "MAX": true,
        "FLAGS": 9,
        "ID": "ERROR_JAMMED",
        "TYPE": "BOOL",
        "DEFAULT": false
      },
```

**Reply by eikowagenknecht** [reply]
Großartig übrigens mit dem Export-Service, das klappt echt gut um solche Daten flott bereitzustellen!

**Reply by eikowagenknecht** [reply]
Ausgelagert aus diesem Thread, siehe https://github.com/danielperna84/hahomematic/discussions/176

**Reply by SukramJ** [reply]
Bitte  tue mir einen gefallen. Das ist ganz schöner Nerdkram, den ich mir mal in Ruhe anschauen muss.
Erstelle dafür bitte einen Issue/Feature Request. Wenn ich das beim WRCD richtig gesehen habe werden die meisten Werte über das MASTER Set bearbeitet, indem die Konfigurationsdaten enthalten sind. In der Integration verwenden wir aber nur das VALUES Set.

Interessant sind aber auch ein paar relevante Usecases die damit verbunden sind. Ich will natürlich nichts versprechen, aber ich werde es mir auf jedenfalls anschauen.

Feature Request wäre gut, dann missbrauchen wir diesen Thread nicht weiter.

**Reply by eikowagenknecht** [reply]
Na klar, kommt gleich in einen eigenen Thread. Für mich persönlich ist eine Lösung davon auch nicht wirklich wichtig, ich habe gerade ein Skript gebaut, dem ich die Zeilen einfach als Variablen reingebe und das dann den passenden Befehl absetzt. Geht mir eher um User, die nicht ganz so technikaffin sind :-)

**Reply by eikowagenknecht** [reply]
DLD ist jetzt an der Terrassentür installiert und funktioniert mit der Integration bestens!

**Comment by SukramJ** [comment]
Meine Geräte, verbunden mit RaspberryMatic (Proxmox VE)

HM-LC-Sw1-FM
HM-LC-Sw4-DR
HM-Sen-MDIR-O
HM-SwI-3-FM
HmIP-ASIR
HmIP-ASIR-2
HmIP-BDT
HmIP-BSL
HmIP-BSM
HmIP-BWTH
HmIP-DRSI4
HmIP-eTRV-2
HmIP-FDT
HmIP-FDT
HmIP-HAP
HmIP-KRCA
HmIP-MIO16-PCB
HmIP-MOD-OC8
HMIP-PS
HMIP-PSM
HmIP-RC8
HmIP-SAM
HmIP-SCI
HmIP-SLO
HmIP-SMI
HmIP-SMO
HmIP-SMO-A
HmIP-SPI
HmIP-SRD
HmIP-SRH
HmIP-STE2-PCB
HmIP-STHD
HmIP-STHO
HmIP-SWD
HMIP-SWDO
HmIP-SWDO-I
HmIP-SWO-PR
HmIP-SWSD
HMIP-WRC2
HmIP-WRC6
HmIP-WRCD
RPI-RF-MOD

**Reply by sVnsation** [reply]
Darf ich fragen, was du für Anwendungsfälle mit dem HmIP-WRCD umgesetzt hast?
Finde das Teil auch interessant und wollte damit immer gern eine Alarmtastatur zur Eingabe der Zahlenkombination realisieren.. mit nur zwei Tastern ist das aber nur beschränkt möglich.

**Reply by SukramJ** [reply]
Die Anwendungsfälle suche ich auch noch, aber eine Alarmtastatur wird es nicht.

**Comment by SukramJ** [comment]
Da ist ja schon mal eine ganz nette Liste an Live testbaren Geräten zusammengekommen. Freut mich.
Das auch die Geräteabdeckung in pydevccu dadurch gewachsen ist freut mich natürlich noch mehr, den dadurch verbessert sich die Testbarkeit, zumindest in Grenzen, erheblich.

**Comment by moelloman** [comment]
Habe folgende Geräte von einer CCU2 erfolgreich integriert:
 HM-Sen-MDIR-O-3
HM-Sec-SCo
HM-Sec-SC-2
HmIP-MOD-HO
HM-RC-4-3
HM-Sec-TiS
HM-LC-Sw4-PCB
HM-LC-Sw4-WM
HM-Sec-SD-2
HM-Sen-RD-O
HM-LC-Sw2-FM
HM-LC-Sw1-FM
HM-WDS30-OT2-SM-2
HM-WDS30-T-O
HmIP-SPI
HmIP-SAM
HMIP-PS
HM-WDS10-TH-O
HM-PBI-4-FM
HM-SwI-3-FM

Besonders happy bin ich das die Multifunktaster (die letzten 2 Modelle) funktionieren. Super Integration! 👍

**Comment by TOBx2** [comment]
Hallo zusammen,

zuerst ein großes Dankeschön an den Projektinitiator und die Unterstützer! Die Standard-HA-Integration ist ja leider etwas …meh; daher super, dass hier eine hoffentlich bessere und aktuellere Alternative entsteht. :)
Leider habe ich mich noch nicht "getraut" auf hahomematic zu wechseln, da mir der Reifegrad noch nicht so klar ist - und die liebe Zeit. Sollte aber ein Gerät in der Liste sein, mit dem ich euch weiterhelfen kann, würde das meine Motivation natürlich steigern. ;)

Nachfolgend meine Liste:

HM-ES-TX-WM
HM-LC-Sw2PBU-FM
HM-Sec-Key
HM-WDS30-OT2-SM-2
HmIP-BSL
HmIP-DLD
HmIP-FALMOT-C12
HmIP-FROLL
HmIP-KRC4
HmIP-MOD-HO
HMIP-PSM
HmIP-SMI
HmIP-SWDO-I
HmIP-WTH-2
RPI-RF-MOD

**Reply by TOBx2** [reply]
Custom Homematic ist zwischenzeitlich installiert und läuft problemlos. Meldet euch also gerne, wenn ihr etwas mit einem der Geräte testen wollt.

**Comment by JoMass** [comment]
Hier mein "Zoo" an Geräten:

   "platform_stats": {
      "binary_sensor": 132,
      "climate": 13,
      "cover": 10,
      "light": 3,
      "lock": 1,
      "select": 2,
      "sensor": 110,
      "switch": 36,
      "hub_sensor": 93,
      "hub_binary_sensor": 43
    },
    "devices": [
      "HM-CC-RT-DN",
      "HM-Dis-EP-WM55",
      "HM-Dis-WM55",
      "HM-ES-PMSw1-DR",
      "HM-ES-PMSw1-Pl",
      "HM-LC-Sw1-DR",
      "HM-LC-Sw1-FM",
      "HM-LC-Sw1-PCB",
      "HM-LC-Sw1-Pl-DN-R1",
      "HM-LC-Sw1PBU-FM",
      "HM-LC-Sw4-SM",
      "HM-LC-Sw4-WM",
      "HM-OU-CF-Pl",
      "HM-OU-CFM-TW",
      "HM-PB-2-WM55",
      "HM-PB-2-WM55-2",
      "HM-PB-4Dis-WM-2",
      "HM-PB-6-WM55",
      "HM-RC-19-SW",
      "HM-RC-4-2",
      "HM-RC-4-3",
      "HM-RC-8",
      "HM-RC-Dis-H-x-EU",
      "HM-RC-Sec4-2",
      "HM-Sec-MDIR",
      "HM-Sec-MDIR-2",
      "HM-Sec-RHS",
      "HM-Sec-SC-2",
      "HM-Sec-SCo",
      "HM-Sec-SD-2",
      "HM-Sec-SD-2-Team",
      "HM-Sec-Sir-WM",
      "HM-Sen-LI-O",
      "HM-Sen-MDIR-O-2",
      "HM-Sen-MDIR-O-3",
      "HM-Sen-MDIR-SM",
      "HM-TC-IT-WM-W-EU",
      "HM-WDC7000",
      "HM-WDS10-TH-O",
      "HM-WDS100-C6-O-2",
      "HM-WDS30-OT2-SM",
      "HM-WDS30-OT2-SM-2",
      "HM-WDS30-T-O",
      "HM-WDS40-TH-I-2",
      "HmIP-DLD",
      "RPI-RF-MOD"
    ]

**Comment by Xsez** [comment]
Ich sehe bisher noch nicht wirklich IP Wired Geräte. Wir renovieren aktuell und haben viel HMIP Wired verbaut.

Ich kann bisher hiermit dienen:

HmIP-MOD-HO
HmIPW-SPI
HmIPW-SMI55
HmIPW-SPI
HmIPW-SPI
HmIPW-SPI
HmIPW-SPI
HmIPW-SPI
HmIPW-SMI55
HmIPW-SMI55
HmIPW-SMI55
HmIP-SCTH230
HmIPW-DRBL4
HmIPW-DRBL4
HmIPW-DRD3
HmIPW-DRD3
HmIPW-DRI32
HmIPW-DRI32
HmIPW-DRS4
HmIPW-DRS4
HmIPW-DRS8
HmIPW-DRS8
HmIPW-FALMOT-C12
HmIP-KRC4
HmIP-eTRV-E
HmIP-eTRV-E
HmIP-eTRV-B
HmIP-eTRV-B
HmIP-CCU3
HmIP-HEATING
HmIP-HEATING
HmIP-HEATING
HmIP-HEATING
HmIP-HEATING
HmIP-RCV-50
HmIPW-DRAP
HmIPW-WRC6
HmIPW-WRC6
HmIP-MP3P
HmIP-SWDO-PL
HmIP-SWDO-I
HmIPW-STH
HmIP-STHO-A
HmIPW-WRC6
HmIPW-BRC2
HmIPW-WRC6
HmIPW-BRC2
HmIPW-BRC2
HmIPW-BRC2
HmIPW-WTH
HmIPW-STHD
HmIPW-WTH
HmIPW-WTH
HmIPW-STHD
HmIPW-WTH
HmIP-SWD

**Comment by ChristophCaina** [comment]
Hier meine aktuelle Geräteliste...
```
"devices": [
      "HMIP-eTRV",
      "HmIP-BDT",
      "HmIP-BROLL",
      "HmIP-BSM",
      "HmIP-RFUSB",
      "HmIP-SLO",
      "HmIP-SRH",
      "HmIP-SWDO-I",
      "HmIP-SWDO-PL",
      "HmIP-SWSD",
      "HmIP-eTRV-2",
      "HmIP-eTRV-C"
    ]
```

Des weiteren wird noch ein HM-ES-TX-WM kommen.
Zudem ist noch das ein oder andere Gerät im Schrank, aber nicht verbaut / im Einsatz, daher hier nicht separat aufgelistet.

**Reply by Buddinski88** [reply]
Ich würde gerne eines meiner Geräte noch in die Runde werfen.

**HM-LC-Sw1-DR** ist ein fest [verdrahteter Schaltaktor](https://de.elv.com/homematic-1-kanal-schaltaktor-im-hutschienengehaeuse-hm-lc-sw1-dr-141378) und diesem wird aktuell noch das Attribut LOWBAT mitgegeben.
Das macht bedingt der Verdrahtung so wenig Sinn. 

Kann das noch entfernt werden? :-)

**Reply by SukramJ** [reply]
Wurde schon entfernt, ist aber noch nicht veröffenlicht.
Ist für derartige Meldungen dennoch das falsche Issue ([siehe](https://github.com/danielperna84/hahomematic/issues/445)).

**Reply by Buddinski88** [reply]
> Wurde schon entfernt, ist aber noch nicht veröffenlicht. Ist für derartige Meldungen dennoch das falsche Issue ([siehe](https://github.com/danielperna84/hahomematic/issues/445)).

Vielen Dank für den Hinweis. Ich hatte mich gewundert wo die Diskussion abgeblieben ist. Werde das in Zukunft berücksichtigen.
